### PR TITLE
chore: [PAYM-2240] Use CSS variables

### PR DESCRIPTION
**Pull request by MekhrangizY**
**PR description:**

**Implemented:**

**Checklist:**

- [x] I ran this code locally
- [ ] I wrote the necessary tests
- [x] My code follows the [style guidelines](http://bit.ly/sd-web-style-guide)
- [x] I followed the [instructions](http://bit.ly/sd-web-pr) to create a pull request

**JIRA card:**

**Should know about:**

### CODE REVIEW

File: packages/forms/src/phone/FormikPhoneField.spec.tsx [hunk 1]
```diff
@@ -2,6 +2,8 @@ import { fireEvent, screen, waitFor } from '@testing-library/react';
 import { renderFormField } from '../__testutils__/renderFormField';
 import { FormikPhoneField } from './FormikPhoneField';
 
+jest.setTimeout(10000);
```
Comment (MekhrangizY) [inline packages/forms/src/phone/FormikPhoneField.spec.tsx:5]: pls remove

File: packages/lab/src/file-list-item/__tests__/FileListItem.spec.tsx [hunk 2]
```diff
@@ -17,7 +17,7 @@ it('checks if delete button dont appear', () => {
   expect(screen.queryByTitle('Delete')).toBeNull();
 });
 
-it('checks if FileListItem renders properly', () => {
+it('checks if FileListItem renders properly 1', () => {
   renderComponent(
```
Comment (MekhrangizY) [inline packages/lab/src/file-list-item/__tests__/FileListItem.spec.tsx:21]: revert this

**Review by tokarevfedor1991 — APPROVED** [review]

File: packages/eslint-plugin/src/rules/no-color-literals.ts [hunk 3]
```diff
@@ -20,6 +20,9 @@ function normalizeHex(hex: string): string {
 const COLOR_KEY_CACHE = new Map<string, string>();
 function findColorName(color: string): undefined | string {
   if (COLOR_KEY_CACHE.size === 0) {
+    for (const [key, hex] of Object.entries(ColorV2)) {
+      COLOR_KEY_CACHE.set(normalizeHex(hex), key);
+    }
     for (const [key, hex] of Object.entries(Color)) {
       COLOR_KEY_CACHE.set(normalizeHex(hex), key);
     }
```
Comment (doniyor2109) [inline packages/eslint-plugin/src/rules/no-color-literals.ts:28]: Lets remove this one

File: packages/ui/src/theme/ThemeProvider.tsx [hunk 4]
```diff
@@ -45,11 +48,70 @@ import {
 import { Color } from './Color';
 import { SuperDispatchTheme } from './SuperDispatchTheme';
 
+const GlobalStyles = createGlobalStyle`
```
Comment (doniyor2109) [inline packages/ui/src/theme/ThemeProvider.tsx:51]: Lets extract it to separate file

File: packages/ui/src/theme/ThemeProvider.tsx [hunk 5]
```diff
@@ -45,11 +48,70 @@ import {
...
+    --sd-teal500:#00678A
+    --sd-teal50:#E3F6FC
+    --sd-silver500:#C4CDD5
+    --sd-silver200:#F3F5F8
+    --sd-silver400:#E1E5EA
+    --sd-dark100:#8F949E
+    --sd-dark300:#5B6371
+    --sd-dark500:#192334
+    --sd-pinkRed:#FE4A49
+  }
```
Comment (doniyor2109) [inline packages/ui/src/theme/ThemeProvider.tsx:78]: Lets add space and semicolon

File: packages/eslint-plugin/src/rules/no-color-literals.ts [hunk 6]
```diff
@@ -44,7 +44,8 @@ function* listColorNames(text: string): Generator<string, void> {
 //
 
 const messages = {
-  suggestColor: 'Use `Color.{{name}}` from "@superdispatch/ui"',
+  suggestColor:
+    'Use `ColorV2.{{name}} or Color.{{name}}` from "@superdispatch/ui"',
```
Comment (doniyor2109) [inline packages/eslint-plugin/src/rules/no-color-literals.ts:48]:
```suggestion
    'Use `ColorV2.{{name}}` from "@superdispatch/ui"',
```

File: packages/forms/package.json [hunk 7]
```diff
@@ -15,7 +15,7 @@
   },
   "dependencies": {
     "@superdispatch/dates": "^0.31.0",
-    "@superdispatch/hooks": "^0.31.0",
+    "@superdispatch/hooks": "^0.30.0",
```
[the remark this answers is not in the document]
Comment (doniyor2109) [reply packages/forms/package.json:18]: Lets make this change in separate PR

File: packages/lab/src/alert/Alert.tsx [hunk 8]
```diff
@@ -3,15 +3,15 @@ import {
   Alert as MaterialAlert,
   AlertProps as MaterialAlertProps,
 } from '@material-ui/lab';
-import { Color } from '@superdispatch/ui';
+import { ColorV2 } from '@superdispatch/ui';
```
Comment (doniyor2109) [inline packages/lab/src/alert/Alert.tsx:6]: Lets rename ColorV2 to ColorNew

File: packages/ui/src/theme/ThemeProviderDark.tsx
Comment (doniyor2109) [inline packages/ui/src/theme/ThemeProviderDark.tsx:1]: Lets rename to GlobalStyles

File: packages/ui/src/theme/ThemeProviderDark.tsx [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+    --sd-teal50: ${ColorLightInternal.Teal50};
+    --sd-silver500: ${ColorLightInternal.Silver500};
+    --sd-silver200: ${ColorLightInternal.Silver200};
+    --sd-silver400: ${ColorLightInternal.Silver400};
+    --sd-dark100: ${ColorLightInternal.Dark100};
+    --sd-dark300: ${ColorLightInternal.Dark300};
+    --sd-dark500: ${ColorLightInternal.Dark500};
+  }
+
+
```
Comment (doniyor2109) [inline packages/ui/src/theme/ThemeProviderDark.tsx:32]:
```suggestion
```

File: packages/ui/src/theme/ThemeProviderDark.tsx [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+    --sd-purple50: ${ColorLightInternal.Purple50};
+    --sd-teal300: ${ColorLightInternal.Teal300};
+    --sd-teal500: ${ColorLightInternal.Teal500};
+    --sd-teal50: ${ColorLightInternal.Teal50};
+    --sd-silver500: ${ColorLightInternal.Silver500};
+    --sd-silver200: ${ColorLightInternal.Silver200};
+    --sd-silver400: ${ColorLightInternal.Silver400};
+    --sd-dark100: ${ColorLightInternal.Dark100};
+    --sd-dark300: ${ColorLightInternal.Dark300};
+    --sd-dark500: ${ColorLightInternal.Dark500};
```
Comment (doniyor2109) [inline packages/ui/src/theme/ThemeProviderDark.tsx:29]: Lets add hyphen betwen numbers as well

File: packages/ui/src/theme/Color.ts [hunk 11]
```diff
@@ -165,10 +165,488 @@ export enum Color {
...
+  Teal50 = '#11333D',
+  Teal300 = '#007EAB',
+  Teal500 = '#31C7F5',
+
+  Yellow50 = '#33200A',
+  Yellow300 = '#E8671C',
+  Yellow500 = '#FCA542',
+}
+
+export enum ColorV2 {
```
Comment (doniyor2109) [inline packages/ui/src/theme/Color.ts:481]: Lets call it `ColorVariables`

File: packages/ui/src/theme/Color.ts [hunk 12]
```diff
@@ -165,10 +165,488 @@ export enum Color {
...
+  /**@deprecated Use Dark100 */
+  Grey100 = '#8F949E',
+  /**@deprecated Use Dark200 */
+  Grey200 = '#6A707C',
+  /**@deprecated Use Dark300 */
+  Grey300 = '#5B6371',
+  /**@deprecated Use Dark400 */
+  Grey400 = '#323C4E',
+  /**@deprecated Use Dark450 */
+  Grey450 = '#222F44',
```
Comment (doniyor2109) [inline packages/ui/src/theme/Color.ts:522]: Lets move these colors to internals

File: packages/ui/src/theme/Color.ts [hunk 13]
```diff
@@ -165,10 +165,488 @@ export enum Color {
   Yellow500 = '#B84807',
 }
 
+export enum ColorLightInternal {
```
Comment (timurhamitov) [inline packages/ui/src/theme/Color.ts:168]: @MekhrangizY Can we add an extended comment at the start of the file with description of what these 4 enums represent and what needs to be changed if someone wants to update one of the colors?

File: packages/ui/src/theme/Color.ts [hunk 14]
```diff
@@ -165,10 +165,488 @@ export enum Color {
...
+  Blue10 = 'rgba(0, 112, 245, 0.1)',
+  Red10 = 'rgba(195, 25, 9, 0.1)',
+
+  //
+  // Legacy Neutral Colors
+  //
+
+  /**@deprecated Use Dark100 */
+  Grey100 = '#8F949E',
+  /**@deprecated Use Dark200 */
```
Comment (timurhamitov) [inline packages/ui/src/theme/Color.ts:202]: @MekhrangizY @doniyor2109 I think we replaced deprecated colors in the design system, maybe we can get rid of them in V2?

File: packages/__testutils__/renderCSS.tsx [hunk 15]
```diff
@@ -1,12 +1,16 @@
...
 import { ReactElement } from 'react';
 
-const colors = new Map<string, string>(
-  Object.entries(Color).map(([k, v]) => [v, `Color.${k}`]),
-);
+const colors = new Map<string, string>();
+Object.entries(Color).map(([k, v]) => {
+  colors.set(v, `Color.${k}`);
+});
+Object.entries(ColorV2).map(([k, v]) => {
```
Comment (normurod72) [inline packages/__testutils__/renderCSS.tsx:11]:
```suggestion
Object.entries(ColorV2).forEach(([k, v]) => {
```

File: packages/ui/src/__tests__/index.spec.ts [hunk 16]
```diff
@@ -89,6 +89,243 @@ it('exposes public api', () => {
...
+        "White50": "rgba(255, 255, 255, 0.5)",
+        "Yellow100": "#FFE494",
+        "Yellow200": "#FFDC6B",
+        "Yellow300": "#E8671C",
+        "Yellow400": "#FFA91F",
+        "Yellow50": "#33200A",
+        "Yellow500": "#FCA542",
+        "Yellow75": "#FFF1C2",
+      },
+      "ColorLightInternal": Object {
```
Comment (doniyor2109) [inline packages/ui/src/__tests__/index.spec.ts:171]: Lets not expose internal colors

File: packages/ui/src/__tests__/index.spec.ts [hunk 17]
```diff
@@ -89,6 +89,243 @@ it('exposes public api', () => {
         "Yellow500": "#B84807",
         "Yellow75": "#FFF1C2",
       },
+      "ColorDarkInternal": Object {
```
Comment (doniyor2109) [inline packages/ui/src/__tests__/index.spec.ts:92]: Lets not expose internal colors

**Review by normurod72 — APPROVED** [review]

**Review by doniyor2109 — APPROVED** [review]

File: packages/lab/src/box/Box.tsx [hunk 18]
```diff
@@ -18,7 +18,7 @@ import { createRuleNormalizer, RuleNormalizer } from '../utils/RuleNormalizer';
 //
 
 function normalizeColor(input: unknown): string | undefined {
-  return isColorProp(input) ? Color[input] : undefined;
+  return isColorProp(input) ? ColorV2[input] : undefined;
```
Comment (MekhrangizY) [inline packages/lab/src/box/Box.tsx:21]: it should be ColorPropV2

File: packages/ui/src/theme/__tests__/ThemeProvider.spec.ts [hunk 19]
```diff
@@ -1,11 +1,18 @@
...
   Object.entries(Color).map(([k, v]) => [v, `Color.${k}`]),
 );
+
+const colorsV2 = new Map<string, string>(
+  Object.entries(ColorV2).map(([k, v]) => [v, `ColorV2.${k}`]),
+);
+
+const combinedColors = new Map([...colors, ...colorsV2]);
+
 const colorRegExp = new RegExp(
```
Comment (MekhrangizY) [inline packages/ui/src/theme/__tests__/ThemeProvider.spec.ts:14]: use forEach instead  of map

**Pull request merged**